### ISSUE TITLE
Missing localization in candidate portal

### ISSUE THREAD

**Issue by hiba-machfej** [talentbeyondboundaries/tbbtalentv2#1413]
I checked from the [test portal](https://tctalent-test.org/candidate-portal) and found these small errors:

- [x] **Langauge section problem.**
  When the website first loads it shows everything corrects. When I go to another section like tasks and then come back to profile the error shows, when I reload it is fixed. 
   All okay:
<img width="1434" alt="Screenshot 2024-10-02 at 11 53 27 AM" src="https://github.com/user-attachments/assets/4541a087-8b3a-41ca-b094-fcf57fc33242">
  Error appears:
  
<img width="1440" alt="Screenshot 2024-10-02 at 11 54 11 AM" src="https://github.com/user-attachments/assets/96d506d1-8e29-4d0a-ab3e-3aecd01d80b0">
<img width="958" alt="Screenshot 2024-10-02 at 11 55 23 AM" src="https://github.com/user-attachments/assets/e4ffd59a-0671-426f-b38d-3e3e4a0d8ad2">

- [ ] Also, some parts are not going back to english when translating to Dari. 
 
<img width="1437" alt="Screenshot 2024-10-02 at 11 55 49 AM" src="https://github.com/user-attachments/assets/c07a4b64-7447-4a00-82fe-4921430e9aeb">

**Comment by hiba-machfej** [comment]
<img width="1203" alt="Screenshot 2024-10-01 at 10 50 23 AM" src="https://github.com/user-attachments/assets/49b220df-3c81-4e78-9dcd-eae65eccf317">
<img width="1178" alt="Screenshot 2024-10-01 at 10 53 56 AM" src="https://github.com/user-attachments/assets/1fde120f-91ab-40d6-997e-cacd0c2e22d0">
<img width="1235" alt="Screenshot 2024-10-01 at 10 53 32 AM" src="https://github.com/user-attachments/assets/1b6f5eff-83d2-40ff-8837-0604552e4df2">

**Comment by sadatmalik** [comment]
TC staging candidate portal:
https://tctalent-test.org/candidate-portal/

**Comment by EhsanEhrari** [comment]
<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBOVl0Qnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--73e397aeb07fb1a5f2dd435b7925ccd13b611da5/image.png" alt="image.png" />

**Comment by EhsanEhrari** [comment]
<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBM1V6Qnc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--a481c77120b92dbccc8345c0d4ed3884e4f79826/image.png" alt="image.png" />

Almost the same bug on candidate-portal/home page

**Comment by EhsanEhrari** [comment]
**Initial Investigation:**
After examining the problem, it was discovered that certain attributes like languageName were missing for the Farsi language in Settings > Translations (General).

**Solution:**
Adding Missing Attributes:
The missing translations for Farsi which are as below, were added under Settings > Translations (General).

**Before:**
home.title = سلام {{اسم}}
card.education.major =تخصص در {{رشته تحصیلی}}
 registration.header.step = مرحله {{فعلی}} از {{تمام}} مراحل
registration.language.label.speak = به چه اندازه شما زبان‌ {{زبان}} را صحبت کرده میتوانید؟
registration.language.label.written = به چه اندازه شما به زبان {{زبان}} نوشته و خوانده میتوانید؟
form.error.minvalue = این بخش باید دارای مقدار {{حد اقل}} باشد 

**Now**
home.title = سلام {{name}}
card.education.major =تخصص در {{major}}
 registration.header.step = مرحله {{current}} از {{total}} مراحل
registration.language.label.speak = به چه اندازه شما زبان‌ {{languageName}} را صحبت کرده میتوانید؟
registration.language.label.written = به چه اندازه شما به زبان {{languageName}} نوشته و خوانده میتوانید؟
form.error.minvalue = این بخش باید دارای مقدار {{min}} باشد

**Comment by EhsanEhrari** [comment]
After fixing or adding the missing attributes, I encountered the issue that @hiba-machfej found. This issue involved navigating to another section (such as "Tasks") and returning to the profile page, which resulted in languageName being displayed instead of its actual value.

**Comment by EhsanEhrari** [comment]
I found this tricky one, when Angular components are re-rendered, the translate pipe typically re-evaluates the value. However, if it fails to do so, this could happen due to the fact that Angular is caching the result of the pipe, especially if it's based on data from a function like getLanguageName(language).

After more investigation I found that functions like getLanguageName(language) can sometimes cause change detection issues in Angular. If the function returns a different value, Angular may not re-evaluate the pipe because the function reference remains the same. A better approach would be to compute the value once in the component and use a variable instead of the function in the template.
